### DISCUSSION TITLE
Issue while flashing eMMC

### DISCUSSION THREAD

**Discussion by Shivaraj199** [polarfire-soc/polarfire-soc-documentation#28] [How to] [answered]
Hello,
I am trying to program the eMMC using the prebuilt yocto. My Issue is when i try to flash the prebuilt yocto.wic 
I can prompt the boot in MMUART0  and my system recognizes the the SOC storage. 

Also I wanted to address an issue, i wanted to check normal power draw of the icicle kit with the linux image which came with the board , when i try to acess the web server I am not able to see the webpage and also not able to ping it

**Comment by andrenemat** [comment]
Do you want to try Rufus instead?

**Comment by hughbreslin** [comment, marked as answer]
Hey @Shivaraj199 

> I promt the during MMUART0 boot and my system recognizes the the SOC storage.

Are you saying it recognizes the updated image and boots it or is it just detecting eMMC / SD card? Could USBImager be getting confused and the drive is actually unmounted properly? Does the drive show up as unmounted after programming / the USBImager error?

> when i try to acess the web server I am not able to see the webpage and also not able to ping it

Is this with the latest (2022.05) release of Yocto or an older version?

**Reply by Shivaraj199** [reply]
it is just detecting the eMMC.
The error shows on the cannot mount the voulme on the disk while I try to write it.
Yes I am trying to flash the latest prebuilt yocto version.
Is i nesscary to update the  Icicle Kit Reference Design too, as i am trying to work with one that came with the board.

**Reply by hughbreslin** [reply]
Ahh ok yes please do update the reference design, the [2022.05 release](https://github.com/polarfire-soc/icicle-kit-reference-design/releases/download/v2022.05/MPFS_ICICLE_KIT_2022_05.zip) just came out and this job file will also update your HSS, I'm not sure if you updated that too but it would do no harm in updating both of them :)

**Reply by Shivaraj199** [reply]
> 

How can I update can I am using Linux can  I simply update it with usbImager?

**Reply by hughbreslin** [reply]
If you installed Libero you have a tool included called FlashPro Express which can be used to re-program the FPGA from a .job file, otherwise you can download a standalone copy of FlashPro Express (here)[https://www.microchip.com/en-us/products/fpgas-and-plds/fpga-and-soc-design-tools/programming-and-debug#Downloads] as part of the programming and debug tools.

**Reply by hughbreslin** [reply]
This [video](https://www.youtube.com/watch?v=Foil5pIRBlM&list=PLtQdQmNK_0DTpi5IB0WcN84fDr_Crn47m&index=3) might help, you just need the provided image section :)

**Reply by Shivaraj199** [reply]
Thank you for the quick reply, I should reprogram the FPGA and after that flash again the yocto-image?

Thank you again

**Reply by hughbreslin** [reply]
Yes give that a try, I'm not entirely sure what design + HSS tag is pre-programmed on your kit so the safest option would be updating them and trying to program again

**Reply by hughbreslin** [reply]
For context this was in response to:

> When i try updating the reference design using flash-pro, it dosnt recognize the kit and the dialogue error from the FPE shows you must connect at-least one programmer.
The question I had is  how the specify the directory where it needs to be programmed. Also what does it mean i need to connect atleast one programmer and why it dosnt recognize my board?

Hmm can  you try running lsusb and see if the flashpro is being detected? I'm not sure if you're using the eFP6 or an external flashpro 5, I have both connected in the output below so you should see either of the Actel devices:


`hugh@Alaska:~$ lsusb`
`Bus 005 Device 033: ID 1514:200b Actel Embedded FP6b`
`Bus 003 Device 014: ID 1514:2008 Actel FlashPro5`

By default root access is required to access the FlashPro, you could try running FPExpress as root if your FlashPro is detected using lsusb.

We have a Libero [document](https://onlinedocs.microchip.com/pr/GUID-FABC58FF-E2CC-4557-BA80-9C03AAFAA2D2-en-US-3/index.html?GUID-9B078B28-0A0C-4F14-A6F1-C80E34A43801) outlining the steps for accessing as a non-root user (so you could launch FPExpress as a standard user), if you need I can share a copy of the scripts mentioned in the doc.

**Reply by Shivaraj199** [reply]
@hughbreslin 
I already configured the both flashpro5 and efp6 and  changed the directory mentioned accordingly in the document 

FP6 already exist in /etc/cyusb.conf.
nothing to do!!!
FP6 already exist in /etc/udev/rules.d/88-cyusb.rules.
nothing to do!!!

When I try lsub 
i still cant detect the programmer 
Any idea where i am going wrong?

**Reply by hughbreslin** [reply]
Are you definitely using the USB embedded programming connector and not the USB OTG connector for the eFP6? It’s the [microUSB port beside the power jack](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/boards/mpfs-icicle-kit-es/icicle-kit-sw-developer-guide/icicle-kit-sw-developer-guide.md#connectors).

Can you also ensure [J24](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/boards/mpfs-icicle-kit-es/updating-icicle-kit/updating-icicle-kit-design-and-linux.md#jumper-settings) is closed?

**Reply by Shivaraj199** [reply]
@hughbreslin  Thank you so much 
I can finally use the FPE to flash reference design, I figured out it was with jumper pins 24. It was not completely closed.

**Reply by Shivaraj199** [reply]
But there is another issue i am facing now when i am prompting the boot to flash  the yocto image eMMC is not recognized 
[415.247579] HSS_Storage_Init(): initialize MMC
[415.253395] HSS_MMCInit(): Attempting to select SDCARD ... Failed
[415.312714] HSS_MMCInit(): Attempting to select eMMC ... Failed
[415.373317] USBDMSC_Start(): FLASH_DRIVE_init() returned false...

Earlier i was able to prompt the boot.

i checked my j34/j43 correctly

**Reply by hughbreslin** [reply]
Hmm there’s two more jumpers you could check J35 should be set at 2.5v and J45 should be set at 1.05v.

**Reply by Shivaraj199** [reply]
My jumpers configurations are correct i cross checked all, But HSS fails to recognize eMMC or SD card

**Reply by Shivaraj199** [reply]
> But there is another issue i am facing now when i am prompting the boot to flash the yocto image eMMC is not recognized [415.247579] HSS_Storage_Init(): initialize MMC [415.253395] HSS_MMCInit(): Attempting to select SDCARD ... Failed [415.312714] HSS_MMCInit(): Attempting to select eMMC ... Failed [415.373317] USBDMSC_Start(): FLASH_DRIVE_init() returned false...
> 
> Earlier i was able to prompt the boot.

This is same error i am having

**Reply by hughbreslin** [reply]
Hmm this is odd, could you attach the full boot log for the HSS? I'm just noticing the time on the attached log is up at 415 seconds of uptime so is it getting stuck in a loop attempting to select / boot or what was it doing before this? 

Its now able to select the eMMC / SD card to program the Linux image right, so this issue is solely selecting the storage when trying to boot?

**Reply by Shivaraj199** [reply]
> Hmm this is odd, could you attach the full boot log for the HSS? I'm just noticing the time on the attached log is up at 415 seconds of uptime so is it getting stuck in a loop attempting to select / boot or what was it doing before this?

-  Yes it is in the loop.


> Its now able to select the eMMC / SD card to program the Linux image right, so this issue is solely selecting the storage when trying to boot?
- No its not able to select eMMC

**Reply by hughbreslin** [reply]
Do you have an SD card you could insert to test if it can detect the SD card itself? Can you please attach the full boot output (even as a text file to save a massive post), I can't really determine much from the cut down output.

**Reply by Shivaraj199** [reply]
> Do you have an SD card you could insert to test if it can detect the SD card itself? Can you please attach the full boot output (even as a text file to save a massive post), I can't really determine much from the cut down output.

I flashed the core-image-minimal-dev-icicle-kit-es  (Prebuilt yocto image) on sd card and  still my board dosn't read i
[hss-log.txt](https://github.com/polarfire-soc/polarfire-soc-documentation/files/8816212/hss-log.txt)
t

**Reply by hughbreslin** [reply]
Hey @Shivaraj199 

Hmm ok this is odd, I'm checking around here to see if anyone has encountered this issue but no luck so far. 

I was able to re-produce it on my end by setting j34/j43 to the 3.3v position / opening them. Its odd its not working for either eMMC / SD, if it was an issue with the eMMC I'd expect the SD card to work.

**Reply by Shivaraj199** [reply]
Hey @hughbreslin  I tried on the different system and i update the reference design again  it  recognizes the emmc.
 I guess there is an issue my system port (Which seems a bit odd as it was recognizing earlier).  
But the problem with detected eMMC is solved at the moment.
Thank you for the help !!!

**Reply by hughbreslin** [reply]
Woohoo! :smile: 

No problem, enjoy playing with the kit!

**Comment by Shivaraj199** [comment]
Hello @hughbreslin ,
I have a small query relating the power draw of the icicle kit,
I am just continuing in this discussion  as I don't want to create new issue. 

I connected the laptops ethernet (laptop is connected with wifi)  with icicle kit is on J2 port. When i try to launch the web-server the server is down. I tried to ping the ip address but the destination Host Unreachable. I tried to manually change the python script but i still face the same issue.

**Reply by hughbreslin** [reply]
Hey @Shivaraj199 have you tried with the Icicle kit connected directly to a router or are you just using a direct connection with a static IP set up on your laptop for the connection?

**Reply by Shivaraj199** [reply]
I tried with static address on my system. 
I am pretty sure with router there shouldn't  be an issue, though I didn't try.
Thank you though!!
